str/cell-type-spec/master_cell_spec_runner.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python3
+
+"""
+This script performs cell type specificity analysis as per Cuomo et al. 2025 procedure.
+
+analysis-runner --dataset tenk10k --access-level test --description "Cell type spec" --output-dir str/cell-spec-cuomo/cell_spec_results master_cell_spec_runner.py
+"""
+
+import numpy as np
+import pandas as pd
+import click
+from cpg_utils import to_path
+from cpg_utils.hail_batch import get_batch, output_path
+
+
+def process_scenario_1_2(gene, chrom, cell_types_without_egene, associatr_path):
+    """
+
+    Processes scenario 1 and 2
+    """
+    scenario_1_2_dict = {}
+    for cell_type in cell_types_without_egene:
+        try:
+            df = pd.read_csv(f'{associatr_path}/{cell_type}/{chrom}/{gene}_100000bp_meta_results.tsv', sep='\t')
```

Comment:
Reading the csv and not using the values is inefficient, especially as this involves making a network request to GCS. You could test for existence of the file instead, assuming that you're not concerned about whether the file parses correctly given that it exists.

Also, if this might be called multiple times for a given gene, then it would be worth caching the result.